mlflow/java/client/src/test/java/org/mlflow/tracking/MlflowContextTest.java

```diff
@@ -0,0 +1,72 @@
+package org.mlflow.tracking;
+
+import static org.mockito.Mockito.*;
+
+import static org.mlflow.api.proto.Service.*;
+
+import org.mlflow.tracking.utils.MlflowTagConstants;
+import org.mockito.ArgumentCaptor;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+public class MlflowContextTest {
+  private static MlflowClient mockClient;
+
+  @AfterMethod
+  public static void afterMethod() {
+    mockClient = null;
+  }
+
+  public static MlflowContext setupMlflowContext() {
+    mockClient = mock(MlflowClient.class);
+    MlflowContext mlflow = new MlflowContext(mockClient);
+    return mlflow;
+  }
+
+  @Test
+  public void testSetExperimentName() {
+    // Will throw if there is no experiment with the same name.
+    {
+      MlflowContext mlflow = setupMlflowContext();
+      when(mockClient.getExperimentByName("experiment-name")).thenReturn(Optional.empty());
+      try {
+        mlflow.setExperimentName("experiment-name");
+        Assert.fail();
+      } catch (IllegalArgumentException expected) {
+      }
+    }
+
+    // Will set experiment-id if experiment is returned from getExperimentByName
+    {
+      MlflowContext mlflow = setupMlflowContext();
+      when(mockClient.getExperimentByName("experiment-name")).thenReturn(
+        Optional.of(Experiment.newBuilder().setExperimentId("123").build()));
+      mlflow.setExperimentName("experiment-name");
+      Assert.assertEquals(mlflow.getExperimentId(), "123");
+    }
+  }
+
+  @Test
+  public void testStartRun() {
+    // Sets the appropriate tags
+    ArgumentCaptor<CreateRun> createRunArgument = ArgumentCaptor.forClass(CreateRun.class);
+    MlflowContext mlflow = setupMlflowContext();
+    mlflow.setExperimentId("123");
+    mlflow.startRun("apple", "parent-run-id");
```

Comment:
Could we also test `startRun` with the Consumer, as well as testing every public method?